### DISCUSSION TITLE
chnage @this.set attribute to anoter type without @

### DISCUSSION THREAD

**Discussion by vahidalvandi** [livewire/livewire#5078] [1. Help] [answered]
hi . i want change type `@this.set('content', html)` to another javascript type like `livewire.set('content', html)` 
rerurn livewire.set is not a function
please  help me .

**Comment by joshhanley** [comment]
@vahidalvandi the issue is `@this` actually expands to `window.livewire.find('{{ $componentid }}')`, so if you want to do that in javascript, you have to write that all out yourself and get the component ID programmatically.

Can you expand on why you want to do this and what you are hoping to achieve from doing it?

Thanks!

**Reply by vahidalvandi** [reply]
I need a way to have a replacement for @this.

**Comment by joshhanley** [comment, marked as answer]
@vahidalvandi I just realised that this discussion and #5089 are about the same issue. We will continue working in the discussion #5089
